src/pages/login/login.ts

```diff
@@ -106,8 +109,27 @@ export class LoginPage extends BasePageComponent {
 
       if (error === AuthenticationError.USER_CANCELLED) {
         this.analytics.logException(error, true);
+        const log: Log = this.createLog(LogType.INFO,
```

Comment:
Looks like we're repeating this a few times. Perhaps a function to handle the creation of log and dispatching called `saveAndDispatchLog` that accepts a message as a param?

Reply:
done